packages/next/src/build/webpack/config/blocks/base.ts

```diff
@@ -76,6 +69,26 @@ export const base = curry(function base(
     config.module = { rules: [] }
   }
 
+  config.plugins ??= []
+  if (config.devtool === 'source-map') {
+    config.plugins.push(
+      new DevToolsIgnorePlugin({
+        // TODO: eval-source-map has different module paths than source-map.
+        // We're currently not actually ignore listing anything.
+        shouldIgnorePath,
+      })
+    )
+  } else if (config.devtool === 'eval-source-map') {
+    // We're using a fork of `eval-source-map`
+    config.devtool = false
+    config.plugins.push(
+      new EvalSourceMapDevToolPlugin({
+        moduleFilenameTemplate: config.output?.devtoolModuleFilenameTemplate,
+        shouldIgnorePath,
+      })
+    )
+  }
+
```

Comment:
Only refactoring (and fixing that `devtoolModuleFilenameTemplate` was ignored.

This makes development easier when you quickly want to switch between `eval-source-map` and `source-map`